### ISSUE TITLE
logits长度不等于total_lengths中长度的求和

### ISSUE THREAD

**Issue by whq-pku** [THUDM/slime#1054]
https://github.com/THUDM/slime/blob/a93386e5130c7bfbbcff3f5d4f255bd0f99b5570/slime/backends/megatron_utils/loss.py#L63

通过打印输入的shape，发现logits的长度并不等于total_lengths中长度的求和，我理解logits是把不同小序列拼在一起凑够max-tokens-per-gpu，所以logits的长度实际应该是拼接的不同小序列的求和，即total_lengths中长度的求和，但实际上logits会偏大些，且偏移量不固定，如:

logits:  torch.Size([1, 16640, 76032])      unconcat_tokens:  [torch.Size([16520])]        total_lengths:  [16520]       response_lengths:  [16384]

logits:  torch.Size([1, 15616, 76032])      unconcat_tokens:  [torch.Size([1820]), torch.Size([6247]), torch.Size([7334])]  total_lengths:  [1820, 6247, 7334] response_lengths:  [1720, 6147, 7091]

请问这是咋回事呢

**Comment by zhuzilin** [comment]
We will pad the total length of the sequence to an multiple of tp_size * 128 for better performance here:
https://github.com/THUDM/slime/blob/6986c4698ada539e42eb258afbc22d6c70656b76/slime/backends/megatron_utils/data.py#L72
